tools/CorrelationTestbed/Test-CorrelationInSandbox.ps1

```diff
@@ -0,0 +1,342 @@
+{
+  Remove-Item -Recurse $ResultsPath -Force
+}
+
+New-Item -ItemType Directory $ResultsPath > $nul
+
+# Close Windows Sandbox
+
+function Close-WindowsSandbox {
+    $sandbox = Get-Process 'WindowsSandboxClient' -ErrorAction SilentlyContinue
+    if ($sandbox) {
+      Write-Host '--> Closing Windows Sandbox'
+
+      $sandbox | Stop-Process
+      $sandbox | Wait-Process -Timeout 30
+
+      Start-Sleep 2
+
+      Write-Host
+    }
+    Remove-Variable sandbox
+}
+
+Close-WindowsSandbox
+
+# Initialize Temp Folder
+
+$tempFolderName = 'CorrelationTestStaging'
+$tempFolder = Join-Path -Path ([System.IO.Path]::GetTempPath()) -ChildPath $tempFolderName
+
+New-Item $tempFolder -ItemType Directory -ErrorAction SilentlyContinue | Out-Null
+
+# Set dependencies
+
+$desktopInSandbox = 'C:\Users\WDAGUtilityAccount\Desktop'
+
+$apiLatestUrl = 'https://api.github.com/repos/microsoft/winget-cli/releases/latest'
+
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$WebClient = New-Object System.Net.WebClient
+
+function Get-LatestUrl {
+  ((Invoke-WebRequest $apiLatestUrl -UseBasicParsing | ConvertFrom-Json).assets | Where-Object { $_.name -match '^Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle$' }).browser_download_url
+}
+
+function Get-LatestHash {
+  $shaUrl = ((Invoke-WebRequest $apiLatestUrl -UseBasicParsing | ConvertFrom-Json).assets | Where-Object { $_.name -match '^Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.txt$' }).browser_download_url
+
+  $shaFile = Join-Path -Path $tempFolder -ChildPath 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.txt'
+  $WebClient.DownloadFile($shaUrl, $shaFile)
+
+  Get-Content $shaFile
+}
+
+# Hide the progress bar of Invoke-WebRequest
+$oldProgressPreference = $ProgressPreference
+$ProgressPreference = 'SilentlyContinue'
+
+$desktopAppInstaller = @{
+  fileName = 'Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'
+  url      = $(Get-LatestUrl)
+  hash     = $(Get-LatestHash)
+}
+
+$ProgressPreference = $oldProgressPreference
+
+$vcLibsUwp = @{
+  fileName = 'Microsoft.VCLibs.x64.14.00.Desktop.appx'
+  url      = 'https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx'
+  hash     = 'A39CEC0E70BE9E3E48801B871C034872F1D7E5E8EEBE986198C019CF2C271040'
+}
+$uiLibsUwp = @{
+    fileName = 'Microsoft.UI.Xaml.2.7.zip'
+    url = 'https://www.nuget.org/api/v2/package/Microsoft.UI.Xaml/2.7.0'
+    hash = "422FD24B231E87A842C4DAEABC6A335112E0D35B86FAC91F5CE7CF327E36A591"
+}
+
+if ($UseDev)
+{
+  $devVCLibsFileName = "Microsoft.VCLibs.x64.14.00.Desktop.appx"
+  $devVCLibsPath = Join-Path ${env:ProgramFiles(x86)} "Microsoft SDKs\Windows Kits\10\ExtensionSDKs\Microsoft.VCLibs.Desktop\14.0\Appx\Retail\x64"
+  $devVCLibsPath = Join-Path $devVCLibsPath $devVCLibsFileName
+
+  Copy-Item -Path $devVCLibsPath -Destination $tempFolder -Force
+
+  $devVCLibsPathInSandbox = Join-Path -Path $desktopInSandbox -ChildPath (Join-Path -Path $tempFolderName -ChildPath $devVCLibsFileName)
+}
+else
+{

```

Comment:
Is something missing here?

Reply:
No, I just didn't indent the entire existing block at the time.